pkg/resource/function/hooks.go

```diff
@@ -114,6 +91,23 @@ func (rm *resourceManager) customUpdateFunction(
 		}
 	}
 
+	switch {
```

Comment:
Please put a code comment in here explaining why you are doing this :) Guarantee we will forget a year from now and say "why is this using a switch statement at all?!"